testnet/stacks-node/src/nakamoto_node/signer_coordinator.rs

```diff
@@ -384,6 +386,17 @@ impl SignerCoordinator {
                         ));
                     }
 
+                    // Check if a new Stacks block has arrived
+                    let (canonical_stacks_tip_ch, canonical_stacks_tip_bh) =
+                        SortitionDB::get_canonical_stacks_chain_tip_hash(sortdb.conn()).unwrap();
+                    let canonical_stacks_tip =
+                        StacksBlockId::new(&canonical_stacks_tip_ch, &canonical_stacks_tip_bh);
+                    info!("SignCoordinator: Current stacks tip: {canonical_stacks_tip}, parent: {parent_block_id}");
+                    if canonical_stacks_tip != parent_block_id {
+                        debug!("SignCoordinator: Exiting due to new stacks tip");
+                        return Err(NakamotoNodeError::StacksTipChanged);
+                    }
```

Comment:
I suspect this is what broke the `allow_reorg_within_first_proposal_burn_block_timing_secs` test.

The miner isn't necessarily mining off of the canonical stacks tip in that case (they mine a child of the stacks block ID that their block commit contains). So, I think the check here needs to be "did the stacks tip change, and is the change in my fork?"

Reply:
Thanks, I was just figuring that one out. Is there an easy to check that?

Reply:
I think you want something like `NakamotoChainState::get_highest_known_block_header_in_tenure()`:

```rust
            let parent_tenure_header =
                NakamotoChainState::get_block_header(chain_state.db(), &self.parent_block_id)?;
            let highest_in_tenure = NakamotoChainState::get_highest_known_block_header_in_tenure(
                &mut chain_state.index_conn(),
                &parent_tenure_header.consensus_hash,
            )?
            .ok_or_else(|| ...)?;
            if highest_in_tenure.block_id() != parent_block_id {
               ..
```

Reply:
Thanks for the tip. Fixed in a420939f766230e4c47b1009e813b5a2a8f3e502.